src/Invictus.Testing.LogicApps/LogicAppAuthentication.cs

```diff
@@ -86,7 +121,12 @@ private static async Task<LogicManagementClient> AuthenticateLogicAppsManagement
 
             AuthenticationResult token = await authContext.AcquireTokenAsync("https://management.azure.com/", credential);
 
-            return new LogicManagementClient(new TokenCredentials(token.AccessToken))
+            return await AuthenticateLogicAppsManagementAsync(subscriptionId, token.AccessToken);
+        }
+
+        private static async Task<LogicManagementClient> AuthenticateLogicAppsManagementAsync(string subscriptionId, string accessToken)
+        {
```

Comment:
Let's add guards on both parameters here, in case other methods such as `AuthenticateLogicAppsManagementAsync`, don't provide a token

Reply:
done!